_posts/en/newsletters/2019-11-13-newsletter.md

```diff
@@ -0,0 +1,367 @@
+    [posted][wuille bech32 workaround] to the Bitcoin-Dev mailing some
+    options for addressing the problem and requested feedback on what
+    options people would prefer to see implemented.  One option would be
+    restricting all current bech32 implementations to rejecting any
+    native segwit addresses that don't result in a 22 or 34 byte
+    scriptPubKey.  Then an upgraded version of bech32 could later be
+    developed with better detection of inserted or deleted characters.
+
+    Many other less critical discussions also resulted from the week's
+    review of taproot, and discussion logs are available ([1][tr meet1],
+    [2][tr meet2]) for anyone interested in the discussion details.
+
+    {:#x-only-pubkeys}
+    In other schnorr/taproot news, Jonas Nick published an [informative
+    blog post][x-only pubkeys] about a recent major change to
+    [bip-schnorr][] and [bip-taproot][] that reduced the size of
+    serialized public keys from 33 bytes to 32 bytes without reducing
+    security. See Newsletters [#59][news59 proposed 32B pubkeys] and
+    [#68][news68 taproot update] for previous discussion of this
+    optimization.
+
+- **Possible privacy leak in the LN onion format:** as described in
+  [BOLT4][], LN uses the [Sphinx][] protocol to communicate payment
+  information between LN nodes.  Olaoluwa Osuntokun [posted][osuntokun
+  sphinx] to the Lightning-Dev mailing list this week about a
+  [recently-published][breaking onion routing] flaw in the original
+  description of Sphinx that may allow a destination node to "deduce a
+  lower bound for the length of the path [back to the source node]."
+  <!-- quote from Osuntokun email -->  The fix is easy: instead of
+  initializing part of an onion packet with zero bytes, random-value
+  bytes are used instead.  Osuntokun created a [PR][lnd-onion]
+  implementing this in the onion library used by LND as well as a
+  [documentation PR][bolts #697] for the BOLTs repository.  Other
+  implementations have also adopted the same change (see the
+  [C-Lightning commit][news72 cl onion] below).
+
+- **LN up-front payments:** the current LN protocol returns all money to
+  the spender if a payment attempt fails or is rejected by the receiver,
+  so routing nodes only receive income if the payment attempt succeeds.
+  However, some new applications are using this costless failure
+  mechanism to send data over LN without paying for the bandwidth they
+  use.  LN designers expected this to happen and have previously spent
+  time thinking about how they could add up-front fees to the
+  network---fees that would be paid to routing nodes whether or not a
+  payment attempt succeeded.
+
+    This week, Rusty Russell started a [thread][russell up-front] on the
+    Lightning-Dev mailing list to discuss proposals for up-front fees.
+    Russell proposed a mechanism that combines fees and hashcash-style
+    proof-of-work to try to prevent nodes from using using the extra
```

Comment:
Wow, I seem to have a real problem with word duplication.  I added a note to my todo list to investigate one of the linters @practicalswift previously recommended to see if it can automatically catch this class of mistakes.